Whiskey/Tasks/Invoke-WhiskeyNodeTask.ps1

```diff
@@ -186,41 +117,18 @@ BuildTasks:
 '@)
         }
 
-        # local version of npm gets removed by 'npm install', so call Get-WhiskeyNPMPath to download it again if necessary
-        $npmPath = Get-WhiskeyNPMPath -NodePath $nodePath -ApplicationRoot $workingDirectory
         foreach( $script in $npmScripts )
         {
             Update-Progress -Status ('npm run {0}' -f $script) -Step ($stepNum++)
-            Write-Timing -Message ('Running script ''{0}''.' -f $script)
-            & $nodePath $npmPath 'run' $script '--scripts-prepend-node-path=auto' $noColorArg 
-            Write-Timing -Message ('COMPLETE')
-            if( $LASTEXITCODE )
-            {
-                Stop-WhiskeyTask -TaskContext $TaskContext -Message ('NPM command `npm run {0}` failed with exit code {1}.' -f $script,$LASTEXITCODE)
-            }
+            Write-WhiskeyTiming -Message ('Running script ''{0}''.' -f $script)
+            Invoke-WhiskeyNpmCommand -Name 'run-script' -ArgumentList $script -NodePath $nodePath -ForDeveloper:$TaskContext.ByDeveloper -ErrorAction Stop
+            Write-WhiskeyTiming -Message ('COMPLETE')
         }
 
         Update-Progress -Status ('nsp check') -Step ($stepNum++)
-        $nodeModulesRoot = Join-Path -Path $nodeRoot -ChildPath 'node_modules'
-        $nspPath = Join-Path -Path $nodeModulesRoot -ChildPath 'nsp\bin\nsp'
-        $npmCmd = 'install'
-        if( (Test-Path -Path $nspPath -PathType Leaf) )
-        {
-            $npmCmd = 'update'
-        }
-
-        Write-Timing -Message ('Installing NSP.')
-        & $nodePath $npmPath $npmCmd 'nsp@2.7.0' '-g'
-        Write-Timing -Message ('COMPLETE')
-        if( -not (Test-Path -Path $nspPath -PathType Leaf) )
-        {
-            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('NSP module failed to install to ''{0}''.' -f $nodeModulesRoot)
-        }
-
-        Write-Timing -Message ('Running NSP security check.')
-        $output = & $nodePath $nspPath 'check' '--output' 'json' 2>&1 |
-                        ForEach-Object { if( $_ -is [Management.Automation.ErrorRecord]) { $_.Exception.Message } else { $_ } } 

```

Comment:
Isn't it the reverse? `$output` should only contain STDOUT. If we include STDERR, won't that be unparseable?

Reply:
I mention this because I had previously removed this special handling and then had to add it back in at a later Whiskey release when I realized it was necessary.

If I remember right, the nsp module outputs the vulnerability information to STDERR which is why we need to use `2>&1`. Then that causes a PowerShell `ErrorRecord` object to be captured in `$output` with the vulnerability JSON stored in the `Exception.Message` property of the object. So the additional piece with `ForEach-Object` pulls the JSON out from the exception message so it can be stored in `$output` and parsed as JSON.

Maybe there was a bug with nsp that has since been fixed that no longer requires this handling.